src/stack.js

```diff
@@ -1,5 +1,60 @@
 'use strict'
 
 export default class Stack {
-  // your code here
+  constructor(instantiator) {
+    if (!(typeof instantiator === 'undefined')) {
+      if (!Array.isArray(instantiator)) {
+        this.items = [instantiator]
+        this.size = 1
+      }
+      if (Array.isArray(instantiator)) {
+        this.items = instantiator
+        this.size = instantiator.length
+      }
+    } else {
+      this.items = []
+      this.size = 0
+    }
+  }
+
+
+  push(item) {
+    if (Array.isArray(item)) {
+      this.size = this.size + item.length
+    }
+    if (!(Array.isArray(item))) {
+      item = [item]
+      this.size ++
+    }
+
+    this.items = [...item, ...this.items]
+    return this.items
+  }
+
+
+  pop() {
+    if (this.length === 0) {
+      return null
+    }
+
+    this.size --
+    return this.items.splice(0, 1)[0]
+  }
+
+
+  peek() {
+    return this.size
```

Comment:
For clarity, consider the use of explicit `if/else` statements, with blocks.  As well, consider making explicit comparisons:
```js
this.size === 0
```